### PR TITLE
DOCKER-275: introduce new variables for communication share-alfresco.

### PR DESCRIPTION
**Pull request by anghelutar**
Currently communication between share and alfresco is done via external route. 
This produces (unnecessary) traffic on the front proxy and (at least in some cases) requests go via Internet.

With this PR it's possible to allow direct communication share-alfresco or share-\<lb-alfresco-nodes\>.

Old variables are kept, because they are needed to generate WebDav links on the details pages in share.

### CODE REVIEW

**Comment by tgeens** [comment]
Question: about the WebDAV (and CMIS?) links - do they look (or can they be made to look at) the `X-Requested-Host` etc headers ... ?

I'm asking this because I like the distinction between internal/external traffic, but I'm wondering if we really need the variables for WebDAV links

**Comment by anghelutar** [comment]
Those links appear in share in the details page:
"Copy this link for the WebDAV URL" (see screenshot)

I suppose that with some effort it is possible to customize share to construct them differently, but that effort is always bigger than expected.

Selection_001

**Comment by tgeens** [comment]
You are right, we're talking about Share here, not Alfresco itself.

**Review by tgeens — APPROVED** [review]

**Comment by anghelutar** [comment]
@vierbergenlars ok for you to merge this?

File: src/main/resources/global/91-init-share.sh [hunk 1]
```diff
@@ -10,11 +10,21 @@ ALFRESCO_PORT=${ALFRESCO_PORT:-8080}
 ALFRESCO_PROTOCOL=${ALFRESCO_PROTOCOL:-http}
 ALFRESCO_CONTEXT=${ALFRESCO_CONTEXT:-alfresco}
 
+ALFRESCO_INTERNAL_HOST=${ALFRESCO_INTERNAL_HOST:-alfresco}
```
Comment (vierbergenlars) [inline src/main/resources/global/91-init-share.sh:13]: Can we default this to `$ALFRESCO_HOST` instead? Or is that a bad idea?

Comment (vierbergenlars) [inline src/main/resources/global/91-init-share.sh:13]: Can we default this to $ALFRESCO_HOST instead? Or is that a bad idea?

**Review by vierbergenlars — CHANGES_REQUESTED** [review]

File: src/main/resources/global/91-init-share.sh (hunk 1 above)
Comment (anghelutar) [reply src/main/resources/global/91-init-share.sh:13]: Imo we want to keep these 2 sets of variables independent.
In single-setup cases, one would leave defaults for INTERNAL variables and would set ALFRESCO_HOST to the DNS name.

Comment (vierbergenlars) [reply src/main/resources/global/91-init-share.sh:13]: Oh okay, I understand.

**Review by vierbergenlars — APPROVED** [review]

**Pull request merged**